clients/python/moondream/cloud_vl.py

```diff
@@ -0,0 +1,161 @@
+                    except json.JSONDecodeError:
+                        continue
+
+    def caption(
+        self,
+        image: Union[Image.Image, EncodedImage],
+        length: Literal["long", "short"] = "long",
+        stream: bool = False,
+        settings: Optional[SamplingSettings] = None,
+    ) -> CaptionOutput:
+        if isinstance(image, EncodedImage):
+            raise ValueError("EncodedImage not supported for cloud inference")
+
+        payload = {
+            "image_url": self._encode_image(image),
+            "length": length,
+            "stream": stream,
+        }
+
+        data = json.dumps(payload).encode("utf-8")
+        req = urllib.request.Request(
+            f"{self.api_url}/caption",
+            data=data,
+            headers={**self.headers, "Content-Type": "application/json"},
+        )
+
+        def generator():
+            for chunk in self._stream_response(req):
+                yield chunk
+
+        if stream:
+            return {"caption": generator()}
+
+        with urllib.request.urlopen(req) as response:
+            result = json.loads(response.read().decode("utf-8"))
+            return {"caption": result.get("caption", "")}
+
+    def query(
+        self,
+        image: Union[Image.Image, EncodedImage],
+        question: str,
+        stream: bool = False,
+        settings: Optional[SamplingSettings] = None,
+    ) -> QueryOutput:
+        if isinstance(image, EncodedImage):
+            raise ValueError("EncodedImage not supported for cloud inference")
+
+        payload = {
+            "image_url": self._encode_image(image),
+            "question": question,
+            "stream": stream,
+        }
+
+        data = json.dumps(payload).encode("utf-8")
+        req = urllib.request.Request(
+            f"{self.api_url}/query",
+            data=data,
+            headers={**self.headers, "Content-Type": "application/json"},
+        )
+
+        if stream:
+            return {"answer": self._stream_response(req)}
+
+        with urllib.request.urlopen(req) as response:
+            result = json.loads(response.read().decode("utf-8"))
+            return {"answer": result.get("answer", "")}
+
+    def detect(
+        self,
+        image: Union[Image.Image, EncodedImage],
+        object: str,
+    ) -> DetectOutput:
+        if isinstance(image, EncodedImage):
+            raise ValueError("EncodedImage not supported for cloud inference")
+
+        payload = {"image_url": self._encode_image(image), "object": object}
+
+        data = json.dumps(payload).encode("utf-8")
+        req = urllib.request.Request(
+            f"{self.api_url}/detect",
+            data=data,
+            headers={**self.headers, "Content-Type": "application/json"},
+        )
+
+        with urllib.request.urlopen(req) as response:
+            result = json.loads(response.read().decode("utf-8"))
+            return {"objects": result.get("objects", [])}
```

Comment:
I think we're better off throwing an error if the API contract is broken and "objects" isn't in the response, instead of defaulting to `[]`. Same for the other capabilities.